torch/nn/attention/flex_attention.py

```diff
@@ -1114,12 +1123,12 @@ def _validate_embed_dim(query: Tensor, key: Tensor, value: Tensor):
 
 
 def _validate_device(query: Tensor, key: Tensor, value: Tensor):
```

Comment:
Add a validate dtype and UT for the dtype float16 case

Reply:
Sure, have added checks specific on [cpu lowering side](https://github.com/pytorch/pytorch/pull/141453/files#diff-7a9af76999e838f26a7171771bae1dd3243aefeed3827a8412f2f19fbf7b8caaR833) since eager mode might still need to run with (like `golden_ref` in test cases).
Also added UT `test_validate_cpu_dtype_error_message`.